builder/config.go

```diff
@@ -26,3 +38,65 @@ type LifecycleConfig struct {
 	URI     string `toml:"uri"`
 	Version string `toml:"version"`
 }
+
+// ReadConfig reads a builder configuration from the file path provided
+func ReadConfig(path string) (Config, error) {
+	builderDir, err := filepath.Abs(filepath.Dir(path))
+	if err != nil {
+		return Config{}, err
+	}
+
+	file, err := os.Open(path)
+	if err != nil {
+		return Config{}, errors.Wrap(err, "opening config file")
+	}
+
+	config, err := parseConfig(file, builderDir)
+	if err != nil {
+		return Config{}, errors.Wrapf(err, "parse contents of '%s'", path)
+	}
+
+	return config, nil
+}
+
+// parseConfig reads a builder configuration from reader and resolves relative buildpack paths using `relativeToDir`
+func parseConfig(reader io.Reader, relativeToDir string) (Config, error) {
```

Comment:
We decided not to export this for now because it felt a little awkward. Can export it later if needed.